src/sync/app.cpp

```diff
@@ -30,29 +32,43 @@ JSON.find(KEY) != JSON.end() ? Optional<RET_TYPE>(JSON[KEY].get<RET_TYPE>()) : r
 namespace realm {
 namespace app {
 
-std::shared_ptr<App> App::app(const std::string app_id,
-                              const realm::util::Optional<App::Config> config)
-{
-    return std::make_shared<App>(app_id, config);
-}
+const static std::string default_base_url = "https://stitch.mongodb.com";
+const static std::string default_base_path = "/api/client/v2.0";
+const static std::string default_app_path = "/app";
+const static std::string default_auth_path = "/auth";
+const static uint64_t    default_timeout_ms = 60000;
+
+App::App(const std::string& app_id, const realm::util::Optional<App::Config>& config)
+    : m_app_id(app_id)
+    {
+    std::string base_url;
+    if (config) {
+        base_url = config->base_url.value_or(default_base_url);
+        m_request_timeout_ms = config->default_request_timeout_ms.value_or(default_timeout_ms);
+
+        if (config->transport) {
+            // TODO: Install custom transport
```

Comment:
[minor] This should overwrite the static generic transport factory that the SDKs would be overwriting. This configuration field is to give users their own opportunity to use their own generic network transport.

Reply:
I've addressed this by making the transport per App now instead of a global member.